src/mesh.cpp

```diff
@@ -3143,6 +3188,12 @@ void LibMesh::set_score_data(const std::string& var_name,
 
 void LibMesh::write(const std::string& filename) const
 {
+  if (adaptive_) {
+    fatal_error(fmt::format(
```

Comment:
Because this operation occurs at the end of the simulation, let's soften this to a warning with an early return from this method.